src/Request/uapi-parser.js

```diff
@@ -301,50 +301,52 @@ Parser.prototype.parseVersion = function (obj) {
 Parser.prototype.parse = function (xml) {
   const self = this;
 
-  return this.parseXML(xml).then((obj) => {
-    const start = new Date();
+  return this
+    .parseXML(xml)
+    .then((obj) => {
+      const start = new Date();
+
+      if (!self.rootObject) {
+        // fall back to custom non-uAPI parser interface with dirty code //FIXME rewrite Hotels
+        return obj;
+      }
 
-    if (!self.rootObject) {
-      // fall back to custom non-uAPI parser interface with dirty code //FIXME rewrite Hotels
-      return obj;
-    }
+      if (obj['SOAP:Fault']) {
+        try {
+          this.uapi_version = this.parseVersion(obj);
+        } catch (e) {
+          throw new RequestRuntimeError.VersionParsingError(obj, e);
+        }
+        return obj;
+      }
 
-    if (obj['SOAP:Fault']) {
+      // trying to redefine version based on response;
       try {
-        this.uapi_version = this.parseVersion(obj);
+        const soapName = self.rootObject.split(':')[0];
+        const rootProps = obj[self.rootObject][0].$;
+        const version = rootProps[`xmlns:${soapName}`].split(`${soapName}_`).pop();
```

Comment:
this code can be separated into function IMO - `extractVersionFromUapi` or whatever and tested